assets/js/lesson_4/user's_portrait.js

```diff
@@ -0,0 +1,221 @@
+// /**
+//  * I know made solution much more complicated then it should be, 
+//  * but I tried to add more checks, though. (I know it's too soon to use while loop)
+//  * If I had more time, I've found more complex solution, maybe :)
+//  */
+
+//Add this variable for check
+
+let valid = false
+
+//Get user birth date.
+
+let userDateInput = Number(prompt
+    (`What is your birth date? \n
+Leave 0 if you don't want tell us your birth date`, 0));
+
+while (!valid) {
+    if (isNaN(userDateInput) || userDateInput >= 2022) {
+        userDateInput = Number(prompt(
+            `You've entered invalid value. Please try again. \n
+        What is your birth date \n
+        Leave o if you don't want tell us your age`, 0));
+    }
+        else if (userDateInput === 0) {
+        alert("What a pity, you didn't add your age!")
+        valid = true
+    } else (valid = true)
+}
+
+//Get user's city
+
+let userCityInput = prompt(
+    `Which city do you live? \n
+Leave this field empty if you don't want tell us where do you live`);
+
+while (!valid) {
+    if (Number(userCityInput)) {
+        userCityInput = prompt(
+            `You've entered invalid value. Please try again. \n
+        Which city do you live? \n
+        Leave this field empty if you don't want tell us where do you live`);
+    }
+        else if (userCityInput === "" || userCityInput === null) {
+        alert("What a pity, you didn't add your city!")
+        valid = true
+    } else (valid = true)
+}
+
+//Get user's fav sport
+
+let userFavSportInput = prompt(
+    `What is you fav sport game? \n
+Leave this field empty if you don't want tell what's your fav sport`);
+
+while (!valid) {
+    if (Number(userFavSportInput)) {
+        userFavSportInput = prompt(
+            `You've entered invalid value. Please try again. \n
+        What is you fav sport game? \n
+        Leave this field empty if you don't want tell what's your fav sport`);
+    }
+        else if (userFavSportInput === "" || userFavSportInput === null) {
+        alert("What a pity, you didn't add your fav sport!")
+        valid = true
+    } else (valid = true)
+}
+
+//Calc age
+
+let userAgeResult;
+
+if ((userDateInput !== 0)) {
+    const currentDate = new Date().getFullYear();
+    const userAgeCalc = currentDate - userDateInput;
+    userAgeResult = `You're ${userAgeCalc} old`
+} else (userAgeResult = "You didn't add age")
+
+// Check capitals
+
+let userCityResult;
+
+if (userCityInput === 'Kyiv') {
```

Comment:
don't forget about typos, it will be better to check all data in lowercase, to avoid the problem with 
userCity = 'kyiv'
userCity = 'KYiv'
...